backend/apps/core/validators.py

```diff
@@ -0,0 +1,95 @@
+"""Validators for the search parameters of the Algolia endpoint."""
+
+from django.core.exceptions import ValidationError
+from django.core.validators import validate_slug
+from django.http import JsonResponse
+
+
+def validate_string_slug(value):
+    """Validate a string value as a slug."""
+    if not value or not isinstance(value, str):
+        return lambda field_name: JsonResponse(
```

Comment:
I don't think this module should operate with JsonResponse. Just return an error message or None upon validation and process it on the upper level.